From 7f39df9735fdb18218819bd0b8a2668463998962 Mon Sep 17 00:00:00 2001
From: Hui Tang <tanghui20@huawei.com>
Date: Wed, 19 Jul 2023 15:04:49 +0800
Subject: [PATCH 35/36] [Huawei] sched: Fix build error for
 dynamic_affinity_enable()

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 189004

-------------------------------

Fix build error if defined(!CONFIG_JUMP_LABEL) &&
defined(CONFIG_QOS_SCHED_DYNAMIC_AFFINITY).

Fixes: eb6843d9adcb ("[Huawei] sched: fix performance degradation on lmbench")
Signed-off-by: Hui Tang <tanghui20@huawei.com>
---
 kernel/sched/fair.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 5a2e14d1462c..395e69bb805c 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -7254,7 +7254,6 @@ static int find_energy_efficient_cpu(struct task_struct *p, int prev_cpu)
 
 #ifdef CONFIG_QOS_SCHED_DYNAMIC_AFFINITY
 
-#ifdef CONFIG_JUMP_LABEL
 static DEFINE_STATIC_KEY_FALSE(__dynamic_affinity_used);
 
 static inline bool dynamic_affinity_used(void)
@@ -7267,13 +7266,6 @@ void dynamic_affinity_enable(void)
 	static_branch_enable_cpuslocked(&__dynamic_affinity_used);
 }
 
-#else /* CONFIG_JUMP_LABEL */
-static bool dynamic_affinity_used(void)
-{
-	return true;
-}
-#endif
-
 /*
  * Low utilization threshold for CPU
  *
-- 
2.17.1

